### PR TITLE
fix: add initContainers to verify if apisix is ready

### PR DESCRIPTION
**Pull request by gxthrj**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### CODE REVIEW

File: docs/en/latest/practices/the-hard-way.md [hunk 1]
```diff
@@ -750,6 +750,10 @@ spec:
             items:
               - key: config.yaml
                 path: config.yaml
+      initContainers:
+        - name: wait-apisix-admin
+          image: busybox:1.28
+          command: ['sh', '-c', "until nc -z apisix-service.apisix.svc.cluster.local 9080 ; do echo waiting for apisix-admin; sleep 2; done;"]
```
Comment (tokers) [inline docs/en/latest/practices/the-hard-way.md:756]: Its port is `9180`.

File: test/e2e/scaffold/ingress.go [hunk 2]
```diff
@@ -214,6 +214,10 @@ spec:
         app: ingress-apisix-controller-deployment-e2e-test
     spec:
       terminationGracePeriodSeconds: 0
+      initContainers:
+      - name: wait-apisix-admin
+        image: localhost:5000/busybox:1.28
+        command: ['sh', '-c', "until nc -z apisix-service-e2e-test.%s.svc.cluster.local 9080 ; do echo waiting for apisix-admin; sleep 2; done;"]
```
Comment (tokers) [inline test/e2e/scaffold/ingress.go:220]: Ditto, admin port is `9180`.

File: test/e2e/scaffold/ingress.go [hunk 3]
```diff
@@ -217,7 +217,7 @@ spec:
       initContainers:
       - name: wait-apisix-admin
         image: localhost:5000/busybox:1.28
-        command: ['sh', '-c', "until nc -z apisix-service-e2e-test.%s.svc.cluster.local 9080 ; do echo waiting for apisix-admin; sleep 2; done;"]
+        command: ['sh', '-c', "until nc -z apisix-service-e2e-test.%s.svc.cluster.local 9180 ; do echo waiting for apisix-admin; sleep 2; done;"]
```
Comment (tokers) [inline test/e2e/scaffold/ingress.go:220]: Here we add another template `%s`, but it seems you don't change the place we render it.

Comment (gxthrj) [reply test/e2e/scaffold/ingress.go:220]: https://github.com/apache/apisix-ingress-controller/pull/660/files#diff-37e06787f8794acb42268c2dde87b4907b54af48e47fdb3b4e2648f97c9fc65aR350 the %s is replaced here.

File: Makefile [hunk 4]
```diff
@@ -104,6 +104,10 @@ ifeq ($(E2E_SKIP_BUILD), 0)
 	docker pull jmalloc/echo-server:latest
 	docker tag  jmalloc/echo-server:latest $(LOCAL_REGISTRY)/jmalloc/echo-server:latest
 	docker push $(LOCAL_REGISTRY)/jmalloc/echo-server:latest
+
+	docker pull busybox:1.28
+	docker tag  busybox:1.28 $(LOCAL_REGISTRY)/busybox:1.28
+	docker push $(LOCAL_REGISTRY)/busybox:1.28
```
Comment (tokers) [inline Makefile:110]: Please don't forget to modify helm chart.

File: test/e2e/scaffold/ingress.go (hunk 3 above)
Comment (tokers) [reply test/e2e/scaffold/ingress.go:220]: OK.

**Review by tokers — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
Let's go forward

**Pull request merged**